erpnext/hr/doctype/leave_application/leave_application.js

```diff
@@ -38,6 +38,9 @@ frappe.ui.form.on("Leave Application", {
 	},
 
 	validate: function(frm) {
+		if (frm.doc.from_date == frm.doc.to_date && frm.doc.half_day == 1){
```

Comment:
Add this validation at python side as well

Reply:
Added in next commit 👍